solidity/contracts/BlockHeaderOracle.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity =0.8.19;
+
+/**
+ * @title BlockHeaderOracle
+ * @notice This contract's purpose is to return the latest stored L1 block header and timestamp
+ * @notice Every X minutes a "magical" off-chain agent provides the latest block header and timestamp
+ */
+contract BlockHeaderOracle {
+  /**
+   * @notice Emits when the block header and timestamp are updated
+   */
+  event BlockHeaderUpdated(bytes _blockHeader, uint256 _blockTimestamp, uint256 _blockNumber);
+
+  /**
+   * @notice The block header
+   */
+  bytes public blockHeader;
+
+  /**
+   * @notice The block timestamp of the latest block header
+   */
+  uint256 public blockTimestamp;
+
+  /**
+   * @notice Updates the block header and timestamp
+   * @param _blockHeader The block header
+   * @param _blockTimestamp The block timestamp
+   * @param _blockNumber The block number
+   */
+  function updateBlockHeader(bytes memory _blockHeader, uint256 _blockTimestamp, uint256 _blockNumber) external {
+    blockHeader = _blockHeader;
+    blockTimestamp = _blockTimestamp;
```

Comment:
Why do we save the timestamp separately? The timestamp can be found inside the header